subprojects/introspect/source/jcli/introspect/groups.d

```diff
@@ -0,0 +1,347 @@
+module jcli.introspect.groups;
+
+/// Mark the member pointers of the command context struct 
+/// to the parent command context struct with this.
+/// That will make it join the command group.
+/// The `onExecute` method will be called after the `onExecute` of that parent executes.
+/// 
+/// For now, when multiple such fields exist, the command  will be a child of both, 
+/// and when the command is resolved, only the context pointer of 
+/// the parent command that it was resolved through will be not-null.
+///
+/// Multiple nested commands in the same function won't work correctly!
+enum ParentCommand;
+
+// TODO: 
+// Ways to specify children instead of the parent.
+// Needs some changes in the graph code. Not hard.
+// TODO: 
+// Ways to parent all commands within a module.
+// I bet this one can be really useful.
+// Should also be pretty easy to implement.
+// TODO:
+// Discover commands throught a parent command, by looking at its children.
+// Should be very useful when you decide to go top-down
+// and have more control over the command layout.
+// I imagine it shouldn't be that hard?
+
+import std.traits;
+import std.meta;
+
+template escapedName(T)
+{
+    import std.string : replace;
+    import std.conv : to;
+    import std.path : baseName;
+
+    // The idea here is to minimize collisions between type symbols.
+    enum location = __traits(getLocation, T);
+    enum escapedName = (baseName(location[0]) ~ fullyQualifiedName!T)
+            .replace(".", "_")
+            .replace(" ", "_")
+            .replace("!", "_")
+        ~ location[1].to!string;
+}
+
+struct TypeGraphNode
+{
+    int childIndex;
+    int fieldIndex;
+}
+
+template TypeGraph(Types...)
+{
```

Comment:
Might be a waste of time, idk, but I wonder if we should make a function to output dot files for [graphviz](https://graphviz.org/) so we can visualise the TypeGraph.

Reply:
You can do that if you want, I'm not the one who's going to do it